cpp/open3d/core/SYCLUtils.h

```diff
@@ -24,14 +24,41 @@
 // IN THE SOFTWARE.
 // ----------------------------------------------------------------------------
 
+/// \file SYCLUtils.h
+/// \brief Common SYCL utilities
+///
+/// SYCLUtils.h and SYCLUtils.cpp should compile when BUILD_SYCL_MODULE=ON or
+/// BUILD_SYCL_MODULE=OFF. Use macros for conditional compilation.
+
 #pragma once
 
+#include <vector>
+
+#include "open3d/core/Device.h"
+
 namespace open3d {
 namespace core {
+namespace sycl_utils {
 
 /// Runs simple SYCL test program for sanity checks.
 /// \return Retuns 0 if successful.
 int SYCLDemo();
```

Comment:
While this seems to be an important function for testing during development. This might not belong to core functionality. 
`IsSYCLWorkingAsExpected` or something like that might be better. Also, return True (boolean) if successful. 
Or can we move this implementation to our `test`.

Reply:
> While this seems to be an important function for testing during development. This might not belong to core functionality.

This is true. We can rename it to have a better name. For CUDA, we are using `open3d::core::cuda`. However, the official SYCL library is already using the `sycl::` namespace, so it will cause confusion if we call it `open3d::core::sycl`. Do you have a suggestion for the namespace name?

> `IsSYCLWorkingAsExpected`. 

This is the `IsAvailable()` function. We are following the Torch's convention: [torch.cuda.is_available](https://pytorch.org/docs/stable/generated/torch.cuda.is_available.html) .

Reply:
For the utils, to be consistent to our user, i think it is the best to use:
```cpp
open3d::core::cuda::xxx();
open3d::core::sycl::xxx();
```

Then, internally, we refer to the official SYCL namespace as:
```cpp
using sy = cl::sycl;
```

Reply:
![image](https://user-images.githubusercontent.com/1501945/177037243-039dc895-d81f-4a93-a05c-afea88a12926.png)

Intel's SYCL implementation is using `namespace s = cl::sycl;`. We can do two letters `sy::` to be more clear.